### PR TITLE
hypershift: add cluster-bot secret

### PR DESCRIPTION
**Pull request by ironcladlou**

### CODE REVIEW

File: core-services/secrets/hypershift-clusterbot/hypershift-clusterbot.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  - csrwng
+  - enxebre
+  - ironcladlou
+  - sjenning
+---
+# this adds the admins to the project.
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: hypershift-clusterbot-viewer-binding
```
Comment (enxebre) [inline core-services/secrets/hypershift-clusterbot/hypershift-clusterbot.yaml:26]: I'd remove binding from the name since it's implicit in the kind

Comment (ironcladlou) [reply core-services/secrets/hypershift-clusterbot/hypershift-clusterbot.yaml]: Fixed

File: core-services/secret-mirroring/_mapping.yaml [hunk 2]
```diff
@@ -566,3 +566,9 @@ secrets:
   to:
     namespace: coreos
     name: coreos-registry-registry.ci.openshift.org
+- from:
+    namespace: hypershift-clusterbot
+    name: admin-credentials
```
Comment (alvaroaleman) [inline core-services/secret-mirroring/_mapping.yaml:571]: This doesn't exist.
/hold

Comment (ironcladlou) [reply core-services/secret-mirroring/_mapping.yaml:571]: Something I don't understand about the secret mirroring — the namespace itself is defined in this PR (`core-services/secrets/hypershift-clusterbot/hypershift-clusterbot.yaml`). What is the order of operations to make this happen? And after the namespace is created (by doing what...?) does that secret have to exist before I commit the mapping? If so, what happens if the secret is ever deleted? etc.

Comment (alvaroaleman) [reply core-services/secret-mirroring/_mapping.yaml:571]: Good point. Yeah, chicken egg problem. The whole mechanism was never explicitly designed but organically grew over time. We are working on a replacment but its not done yet.
/hold cancel
/lgtm
/approve

**Pull request merged**